### PR TITLE
Boot providers during tests

### LINKED ISSUE #153: code in providers boot() not called during the tests
That's not a big deal as more often it is used as part of the lifecycle request but I can see some use cases:
- during unit tests
- in somme console commands

For example, I was trying to test View facade to render a view. We might have to do this in a console command or to assess behaviour in a unit test, and here as boot() methods of providers are not called, some stuff are not shared in the view that are usually shared (session, request,...) and that can lead to errors.

Should be boot providers during tests if we are not making a request ?
What's your opinon on that ?

### PR DESCRIPTION
**Pull request by girardinsamuel**
Fixes #153.

Here I don't load the RouteProvider during tests as we don't have a CSRF token yet. Of course it will be loaded when doing a request such as `self.get()` or an other method.

I added a test on view which is testing that helpers are correctly shared in views. They are shared during boot of ViewProvider, so this a proof that it is working 😉

### CODE REVIEW

File: src/masonite/providers/SessionProvider.py [hunk 1]
```diff
@@ -13,10 +13,9 @@ def register(self):
         session = Session(self.application).set_configuration(config("session.drivers"))
         session.add_driver("cookie", CookieDriver(self.application))
         self.application.bind("session", session)
-        self.application.make("view").share({"old": self.old})
 
     def boot(self):
-        pass
+        self.application.make("view").share({"old": self.old})
```
Comment (josephmancuso) [inline src/masonite/providers/SessionProvider.py:18]: why is this moved into the boot?

Comment (girardinsamuel) [reply src/masonite/providers/SessionProvider.py:18]: I don't know lol I was thinking that it was the correct place to define view sharing...

Comment (girardinsamuel) [reply src/masonite/providers/SessionProvider.py:18]: do you want to revert this ?

Comment (josephmancuso) [reply src/masonite/providers/SessionProvider.py:18]: I don't think it is right? Well it depends. In this case the old method is not called until you call it in your templates. 

In this case it will make each request slower because it has to register this old method every request. 

If we keep it in the register method it only gets registered once and it should still work

File: src/masonite/tests/TestCase.py [hunk 2]
```diff
@@ -25,6 +27,18 @@ def setUp(self):
             self.startTestRun()
         self.withoutCsrf()
         self._exception_handling = False
+        # boot providers as they won't not be loaded if the test is not doing a request
+        self.application.bind("environ", generate_wsgi())
+        try:
+            for provider in self.application.get_providers():
+                # if no request is made we don't need RouteProvider, and we can't load it anyway
+                # because we don't have created a CSRF token yet
+                if not isinstance(provider, RouteProvider):
```
Comment (josephmancuso) [inline src/masonite/tests/TestCase.py:36]: Also i thought we get around the csrf checks by setting a cookie and the csrf input?

Comment (josephmancuso) [reply src/masonite/tests/TestCase.py:36]: hmm not sure what the best approach is but if this works i'll merge it in and we can revisit it later

**Review by josephmancuso — APPROVED** [review]
